scipy/signal/_signaltools.py

```diff
@@ -2452,6 +2456,282 @@ def hilbert2(x, N=None):
     return x
 
 
+def envelope(z: np.ndarray, bp_in: tuple[int | None, int | None] = (1, None), *,
+             n_out: int | None = None, squared: bool = False,
+             residual: Literal['lowpass', 'all', None] = 'lowpass',
+             axis: int = -1) -> np.ndarray:
+    r"""Compute the envelope of a real- or complex-valued signal.
+
+    Parameters
+    ----------
+    z : ndarray
+        Real- or complex-valued input signal, which is assumed to be made up of ``n``
+        samples and having sampling interval ``T``. `z` may also be a multidimensional
+        array with the time axis being defined by `axis`.
+    bp_in : tuple[int | None, int | None]
+        2-tuple defining the frequency band ``bp_in[0]:bp_in[1]`` of the input filter.
+        The corner frequencies are specified as integer multiples of ``1/(n*T)`` with
+        ``-n//2 <= bp_in[0] < bp_in[1] <= (n+1)//2`` being the allowed frequency range.
+        ``None`` entries are replaced with `-n//2` or `(n+1)//2` respectively. The
+        default of ``(1, None)`` removes the mean value as well as the negative
+        frequency components.
+    n_out : int | None
+        If not ``None`` the output will be resampled to `n_out` samples. The default
+        of ``None`` sets the output to the same length as the input `z`.
+    squared : bool
+        If set, the square of the envelope is returned. The bandwidth of the squared
+        envelope is often smaller than the non-squared envelope bandwidth due to the
+        nonlinear nature of the utilized absolute value function. I.e., the embedded
+        square root function typically produces addiational harmonics.
+    residual : Literal['lowpass', 'all', None]
+        This option determines what kind of residual, i.e., the signal part which the
+        input bandpass filter removes, is returned. ``'all'`` returns everything except
+        the contents of the frequency band ``bp_in[0]:bp_in[1]``, ``'lowpass'``
+        returns the contents of the frequency band ``< bp_in[0]``. If ``None`` then
+        only the envelope is returned.
+    axis : int
+       Axis of `z` over which to compute the envelope. Default is last the axis.
+
+    Returns
+    -------
+    np.ndarray
```

Comment:
for consistency with above
```suggestion
    ndarray
```